extensions/data-transfer/portability-data-transfer-generic/src/main/java/org/datatransferproject/datatransfer/generic/GenericTransferExtension.java

```diff
@@ -0,0 +1,214 @@
+import static org.datatransferproject.types.common.models.DataVertical.VIDEOS;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.datatransferproject.api.launcher.ExtensionContext;
+import org.datatransferproject.spi.cloud.storage.AppCredentialStore;
+import org.datatransferproject.spi.cloud.storage.JobStore;
+import org.datatransferproject.spi.transfer.extension.TransferExtension;
+import org.datatransferproject.spi.transfer.provider.Exporter;
+import org.datatransferproject.spi.transfer.provider.Importer;
+import org.datatransferproject.types.common.models.DataVertical;
+import org.datatransferproject.types.common.models.blob.BlobbyStorageContainerResource;
+import org.datatransferproject.types.common.models.calendar.CalendarContainerResource;
+import org.datatransferproject.types.common.models.media.MediaContainerResource;
+import org.datatransferproject.types.common.models.social.SocialActivityContainerResource;
+import org.datatransferproject.types.transfer.auth.AppCredentials;
+import org.datatransferproject.types.transfer.serviceconfig.TransferServiceConfig;
+
+class GenericTransferServiceVerticalConfig {
+  private final DataVertical vertical;
+
+  @JsonCreator
+  public GenericTransferServiceVerticalConfig(
+      @JsonProperty(value = "vertical", required = true) DataVertical vertical) {
+    this.vertical = vertical;
+  }
+
+  public DataVertical getVertical() {
+    return vertical;
+  }
+}
+
+class GenericTransferServiceConfig {
+  private final String serviceId;
+  private final URL endpoint;
+  private final List<GenericTransferServiceVerticalConfig> verticals;
+
+  public GenericTransferServiceConfig(
+      @JsonProperty(value = "serviceId", required = true) String serviceId,
+      @JsonProperty(value = "endpoint", required = true) URL endpoint,
+      @JsonProperty(value = "verticals", required = true)
+          List<GenericTransferServiceVerticalConfig> verticals) {
+    this.serviceId = serviceId;
+    this.endpoint = endpoint;
+    this.verticals = verticals;
+  }
+
+  public String getServiceId() {
+    return serviceId;
+  }
+
+  public URL getEndpoint() {
+    return endpoint;
+  }
+
+  public List<GenericTransferServiceVerticalConfig> getVerticals() {
+    return verticals;
+  }
+
+  public boolean supportsVertical(DataVertical vertical) {
+    for (GenericTransferServiceVerticalConfig verticalConfig : verticals) {
+      if (verticalConfig.getVertical().equals(vertical)) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
nitpick: Probably it's worth building a set of Verticals from `verticals` attribute and do a `Set.contains`